#################################################################
#Population Genomic Analysis pipeline for Mimulus cardinalis

#Author: Daniel Anstett
#Last updated Feb 17 2021
#################################################################


#################################################################
#Filter Illumina adapters & Trim Low Quality



#!/bin/bash
 
#PBS -l walltime=05:00:00,select=1:ncpus=30:mem=32gb
#PBS -N 12_trim
#PBS -A st-angert-1
#PBS -m abe
#PBS -M daniel.anstett@botany.ubc.ca
#PBS -o 12_trim_pigz.output.txt
#PBS -e 12_trim_pigz.error.txt
 
################################################################################

source ~/miniconda3/etc/profile.d/conda.sh
conda activate trim

cd $PBS_O_WORKDIR

FILE=/project/st-angert-1/12_test/12_test.csv

while IFS=, read -r f1; 
do

trimmomatic PE /project/st-angert-1/12_test/$f1"_R1.fastq.gz" /project/st-angert-1/12_test/$f1"_R2.fastq.gz" -threads 30 /scratch/st-angert-1/output_12/$f1"_R1_paired.fastq" /scratch/st-angert-1/output_12/$f1"_R1_unpaired.fastq" /scratch/st-angert-1/output_12/$f1"_R2_paired.fastq" /scratch/st-angert-1/output_12/$f1"_R2_unpaired.fastq" ILLUMINACLIP:/project/st-angert-1/raw_data/adaptors_illumina.fasta:2:30:10:8:T SLIDINGWINDOW:4:15 MINLEN:36 
 
pigz /scratch/st-angert-1/output_12/$f1"_R1_paired.fastq" -p 30
pigz /scratch/st-angert-1/output_12/$f1"_R2_paired.fastq" -p 30
pigz /scratch/st-angert-1/output_12/$f1"_R1_unpaired.fastq" -p 30
pigz /scratch/st-angert-1/output_12/$f1"_R2_unpaired.fastq" -p 30

done < $FILE

conda deactivate

