#################################################################
#Population Genomic Analysis pipeline for Mimulus cardinalis

#Author: Daniel Anstett
#Last updated May 20  2021
#################################################################


#################################################################
# Extract indel coverage chrXX 



#!/bin/bash
 
#PBS -l walltime=01:20:00,select=1:ncpus=1:mem=100gb
#PBS -N hard_filter_indel_chrXX
#PBS -A st-angert-1
#PBS -m abe
#PBS -M daniel.anstett@botany.ubc.ca
#PBS -o hard_filter_indel_chrXX.output.txt
#PBS -e hard_filter_indel_chrXX.error.txt
 
################################################################################

source ~/miniconda3/etc/profile.d/conda.sh
conda activate gatk4

cd $PBS_O_WORKDIR

gatk VariantFiltration -V /scratch/st-angert-1/gold_set/indel/gold_chrXX_indels.vcf.gz --filter-expression "MQ < 50.0" --filter-name "mapping_quality" --filter-expression "AN < 21.6" --filter-name "cov_chr" --filter-expression "SOR < -1.0" --filter-name "strand_odds_ratio_-1" --filter-expression "SOR > 1.0" --filter-name "strand_odds_ratio_1" --filter-expression "AF < 0.25" --filter-name "minor_allele_freq_min" --filter-expression "AF > 0.75" --filter-name "minor_allele_freq_max" --filter-expression "ExcessHet > 10" --filter-name "Excess_H" --filter-expression "BaseQRankSum < -1.0" --filter-name "BaseQRankSum_-1" --filter-expression "BaseQRankSum > 1.0" --filter-name "BaseQRankSum_1" --filter-expression "DP > 211.8844" --filter-name "Upper_DP" --filter-expression "DP < 21.92176 " --filter-name "Lower_DP" --missing-values-evaluate-as-failing true -O /scratch/st-angert-1/gold_set/filtered_indel/gold_chrXX_indels_filtered.vcf.gz

conda deactivate


