#################################################################
#Population Genomic Analysis pipeline for Mimulus cardinalis

#Author: Daniel Anstett
#Last updated June 07 2021
#################################################################


#################################################################
# Select snp variants chrXX 



#!/bin/bash
 
#PBS -l walltime=24:00:00,select=1:ncpus=1:mem=170gb
#PBS -N select_snp_vqsr
#PBS -A st-angert-1
#PBS -m abe
#PBS -M daniel.anstett@botany.ubc.ca
#PBS -o select_snp_vqsr.output.txt
#PBS -e select_snp_vqsr.error.txt
 
################################################################################

source ~/miniconda3/etc/profile.d/conda.sh
conda activate gatk4

cd $PBS_O_WORKDIR

gatk SelectVariants -V /scratch/st-angert-1/10_gatherVCF/all_var_snp.vcf.gz -R /scratch/st-angert-1/mc_ref/CE10g_v2.0.fa -select-type SNP -O /scratch/st-angert-1/10_select_final_snps/all_vqsr_filtered_variants.vcf.gz --restrict-alleles-to BIALLELIC -select "AF > 0.01 && AF < 0.99" -select "AN > 665" -select "vc.isNotFiltered() || vc.getFilters().contains('VQSRTrancheSNP50.00to70.00') || vc.getFilters().contains('VQSRTrancheSNP70.00to90.00')"

gatk IndexFeatureFile -I /scratch/st-angert-1/10_select_final_snps/all_vqsr_filtered_variants.vcf.gz

conda deactivate


